manifests/timer_wrapper.pp

```diff
@@ -0,0 +1,92 @@
+# @param on_calendar the calendar event expressions time to run the service
+# @param service_overrides override for the`[Service]` section of the service
+#   example: `'service_overrides' => { 'Wants' => 'network-online.target' }`
+# @param timer_overrides override for the`[Timer]` section of the timer
+#   example: `'timer_overrides' => { 'OnBootSec' => '10' }`
+# @param service_unit_overrides override for the`[Unit]` section of the service
+#   example: `'service_unit_overrides' => { 'Wants' => 'network-online.target' }`
+# @param timer_unit_overrides override for the `[Unit]` section of the timer
+#   can not be used in combination with `additional_timer_params`
+define systemd::timer_wrapper (
+  Enum['present', 'absent']                 $ensure,
+  Optional[String[1]]                       $command = undef,
+  Optional[String[1]]                       $user = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_active_sec = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_boot_sec = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_start_up_sec = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_unit_active_sec = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_unit_inactive_sec = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_calendar = undef,
+  Systemd::Unit::Service                    $service_overrides         = {},
+  Systemd::Unit::Timer                      $timer_overrides           = {},
+  Systemd::Unit::Unit                       $timer_unit_overrides      = {},
+  Systemd::Unit::Unit                       $service_unit_overrides    = {},
+) {
+  $_timer_spec = {
+    'OnActiveSec'       => $on_active_sec,
+    'OnBootSec'         => $on_boot_sec,
+    'OnStartUpSec'      => $on_start_up_sec,
+    'OnUnitActiveSec'   => $on_unit_active_sec,
+    'OnUnitInactiveSec' => $on_unit_inactive_sec,
+    'OnCalendar'        => $on_calendar,
+  }.filter |$k, $v| { $v =~ NotUndef }
+
+  if $ensure == 'present' {
+    if $_timer_spec == {} {
+      fail('At least one of on_active_sec,
+        on_boot_sec,
+        on_start_up_sec,
+        on_unit_active_sec,
+        on_unit_inactive_sec,
+        or on_calendar must be set'
+      )
+    }
+    if ! $command {
+      fail('command must be set')
+    }
+  }
+
+  $service_ensure = $ensure ? { 'absent' => false,  default  => true, }
+  $unit_name = regsubst($title, '/' , '_', 'G')
+
+  systemd::manage_unit { "${unit_name}.service":
+    ensure        => $ensure,
+    unit_entry    => $service_unit_overrides,
+    service_entry => {
+      'ExecStart' => $command, # if ensure present command is defined is checked above
+      'User'      => $user, # defaults apply
+      'Type'      => 'oneshot',
+    }.filter |$key, $val| { $val =~ NotUndef } + $service_overrides,
+  }
+  systemd::manage_unit { "${unit_name}.timer":
+    ensure      => $ensure,
+    unit_entry  => $timer_unit_overrides,
```

Comment:
This will be undef if there is no `$timer_unit_overrides`  . I think would be good to always have a 
```config
[Unit]
Description=Puppet Generated Timer from $title
```
or something.

Reply:
See https://github.com/voxpupuli/puppet-systemd/pull/419#discussion_r1510351683